src/python/gudhi/point_cloud/timedelay.py

```diff
@@ -0,0 +1,94 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Martin Royer, Yuichi Ike, Masatoshi Takenouchi
+#
+# Copyright (C) 2020 Inria, Copyright (C) 2020 Fujitsu Laboratories Ltd.
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+
+class TimeDelayEmbedding:
+    """Point cloud transformation class.
+    Embeds time-series data in the R^d according to Takens' Embedding Theorem
+    and obtains the coordinates of each point.
+    Parameters
+    ----------
+    dim : int, optional (default=3)
```

Comment:
Question: should we name this `dim` or something like `repeat` or `copies` (or better, I am not super fond of those 2 names)? For scalar time series, it is the same. But we will probably generalize this function later to accept vector time series, in which case, if we start from a 2D time series, we can embed it in dimension 6 by taking 3 *copies* with delays. In that case, would users rather pass 6 or 3 as argument? I don't know, I think I would more naturally pass 3, but both seem ok, maybe we can stick to `dim`.

Reply:
**Name suggestions**
How about in the following order?  
1. replicate
2. recursion
3. iteration
4. rebuild
5. recast

Reply:
I suggest as follows.
What do you think?
```python
plane : int, optional (default=3)
        Determination of delayed coordinate system for space.
```

Reply:
That doesn't really clarify things. Let's stick to `dim`, as you currently have, it is clear that it is the dimension of the space we are embedding into (and not the number of repetitions).